pkg/controller/daemon/daemon_controller.go

```diff
@@ -801,6 +801,70 @@ func (dsc *DaemonSetsController) resolveControllerRef(namespace string, controll
 	return ds
 }
 
+// syncPodOnNode syncs the DaemonSet Pods on the node, the return values are:
+//   - nodesNeedingDaemonPods: the pods need to start on the node
+//   - podsToDelete: the Pods need to be deleted on the node
+//   - failedPodsObserved: the number of failed pods on node
+//   - err: unexpected error
+func (dsc *DaemonSetsController) syncPodOnNode(
+	node *v1.Node,
+	nodeToDaemonPods map[string][]*v1.Pod,
+	ds *extensions.DaemonSet,
```

Comment:
You only need to pass `node.Name`, `nodeToDaemonPods[node.Name]` and `dsKey`

Reply:
seems not change on # of arg, so I keep it :)